rdtools/soiling.py

```diff
@@ -0,0 +1,518 @@
+''' Soiling Module
+
+This module contains functions to calculate soiling
+metrics from photovoltaic system data.
+'''
+
+from __future__ import division
+import pandas as pd
+import numpy as np
+from scipy.stats.mstats import theilslopes
+
+
+# Custom exception
+class NoValidIntervalError(Exception):
+    '''raised when no valid rows appear in the result dataframe'''
+    pass
+
+
+class srr_analysis():
+    '''
+    Class for running the stochastic rate and recovery (SRR) photovoltaic soiling loss
+    analysis presented in Deceglie et al. JPV 8(2) p547 2018
+    '''
+    def __init__(self, daily_normalized_energy, daily_insolation, precip=None):
+        self.pm = daily_normalized_energy  # daily performance metric
+        self.insol = daily_insolation
+        self.precip = precip  # daily precipitation
+
+        if self.pm.index.freq != 'D':
```

Comment:
Are you sure you want to make this raise an exception instead of a warning?  I think it should work if the index is daily even if the `freq` is not set.

Also a docstring on __init__ defining the specifications of the input dataframes is needed

Reply:
Thanks for catching the docstring. 

I'm not sure about raising a warning instead of an error, people frequently blow by warnings and I don't see it as a hardship to make sure the `freq` is set. Calling everything else with non-daily series could lead to garbage results. I've opened an issue so the community can weigh in.